packages/components/legacy_text/src/legacy_text.tsx

```diff
@@ -26,17 +26,18 @@ export type LegacyTextVariant =
   | HeadingVariant
   | SubheadingVariant;
 
-interface OwnProps {
-  variant?: LegacyTextVariant;
-}
-
 type HeadingVariantProps = { variant: HeadingVariant } & Omit<HeadingProps, 'size'>;
 type SubheadingVariantProps = { variant: SubheadingVariant } & Omit<SubheadingProps, 'size'>;
 type BodyVariantProps = { variant: BodyVariant } & Omit<BodyProps, 'size'>;
+type NoVariantProps = Omit<BodyProps, 'size'>;
 
-export type TextProps = (HeadingVariantProps | SubheadingVariantProps | BodyVariantProps) & OwnProps;
+export type TextProps = HeadingVariantProps | SubheadingVariantProps | BodyVariantProps | NoVariantProps;
 
 export const LegacyText: React.FC<TextProps> = (props): React.ReactElement => {
+  if (!("variant" in props)) {
```

Comment:
tried to put this condition the `switch` statement, but since it is a union type, then there is a type error whenever you type `props.variant` until this `'variant' in props` check is performed